app/components/documents/document-card.js

```diff
@@ -138,7 +140,9 @@ export default class DocumentsDocumentCardComponent extends Component {
     return (
       !this.args.hideUpload
         && (!this.hasSignFlow
-            || (this.hasMarkedSignFlow && !!this.args.decisionActivity))
+            || ((this.hasMarkedSignFlow || this.hasSignPreparationActivity || !!this.signCompletionActivity ) 
```

Comment:
This doesn't make sense. A signFlow with a preparationActivity will always have a markingActivity, so this statement will never reach the 2nd OR.